src/systems/interactions.js

```diff
@@ -0,0 +1,199 @@
+    await waitForEvent("body-loaded", entity);
+    state.spawning = false;
+    entity.object3D.position.copy(data.useCustomSpawnPosition ? data.spawnPosition : superSpawner.el.object3D.position);
+    if (data.centerSpawnedObject) {
+      entity.body.position.copy(options.entity.object3D.position);
+    }
+    entity.object3D.scale.copy(data.useCustomSpawnScale ? data.spawnScale : superSpawner.el.object3D.scale);
+    entity.object3D.matrixNeedsUpdate = true;
+  },
+
+  init: function() {
+    this.options = {
+      leftHand: {
+        entity: document.querySelector("#player-left-controller"),
+        grabPath: paths.actions.leftHand.grab,
+        dropPath: paths.actions.leftHand.drop,
+        constraintTag: "offersHandConstraint",
+        hoverFn: findHandCollisionTargetForHand
+      },
+      rightHand: {
+        entity: document.querySelector("#player-right-controller"),
+        grabPath: paths.actions.rightHand.grab,
+        dropPath: paths.actions.rightHand.drop,
+        constraintTag: "offersHandConstraint",
+        hoverFn: findHandCollisionTargetForHand
+      },
+      rightRemote: {
+        entity: document.querySelector("#cursor"),
+        grabPath: paths.actions.cursor.grab,
+        dropPath: paths.actions.cursor.drop,
+        constraintTag: "offersRemoteConstraint",
+        hoverFn: this.getRightRemoteHoverTarget
+      }
+    };
+    this.state = {
+      leftHand: {
+        hovered: null,
+        held: null,
+        spawning: null
+      },
+      rightHand: {
+        hovered: null,
+        held: null,
+        spawning: null
+      },
+      rightRemote: {
+        hovered: null,
+        held: null,
+        spawning: null
+      }
+    };
+
+    this.cursorController = document.querySelector("#cursor-controller");
+  },
+
+  getRightRemoteHoverTarget() {
+    return this.rightRemoteHoverTarget;
+  },
+
+  tickInteractor(options, state) {
+    const userinput = AFRAME.scenes[0].systems.userinput;
+    if (state.held) {
+      const networked = state.held.components["networked"];
+      const lostOwnership = networked && networked.data && networked.data.owner !== NAF.clientId;
+      if (userinput.get(options.dropPath) || lostOwnership) {
+        state.held = null;
+      }
+    } else {
+      state.hovered = options.hoverFn.call(this, options.entity.body);
+      if (state.hovered) {
+        if (userinput.get(options.grabPath)) {
+          const offersConstraint =
+            state.hovered.components.tags && state.hovered.components.tags.data[options.constraintTag];
+          const isHoldableButton = state.hovered.components.tags && state.hovered.components.tags.data.holdableButton;
+          const superSpawner = state.hovered.components["super-spawner"];
+          if (offersConstraint || isHoldableButton) {
+            if (
+              this.el.is("frozen") ||
+              !state.hovered.components.pinnable ||
+              !state.hovered.components.pinnable.data.pinned
```

Comment:
This could be implemented in `pinnable` if it knew to toggle a tag called `holdingBlocked`. I would say that it could toggle `holdable`, but the thought of more than one thing wanting to toggle `holdable` gives me cold sweats at night.